lib/binary/encoder.go

```diff
@@ -19,6 +19,13 @@ type Encoder struct {
 	scratch [binary.MaxVarintLen64]byte
 }
 
+func (enc *Encoder) Nullable() error {
+	if _, err := enc.output.Write([]byte{0}); err != nil {
+		return err
+	}
+	return nil
```

Comment:
This can just be `err := Write(...)` and then `return err` without the `if`.

Reply:
That is how everywhere else is done in the file, so I'm keeping consistency by being more explicit what we return..